mrrc/pkgs/maven.py

```diff
@@ -238,6 +238,17 @@ def handle_maven_uploading(
         )
         logger.info("maven-metadata.xml uploading done")
 
+    # this step generates index.html for each dir and add them to file list
+    index_files = valid_paths
```

Comment:
These invalid_paths and meta_files are just for local entries. If you just use them you will miss entries in s3.
For example, in s3 we have /org/apache/maven/httpclient/{1.3, 1.4}/xxxx, and this time we will just upload 1.5, that means the valid_paths just contains 1.5 entries. If you use them to generate index, all 1.3, 1.4 will be missed.
So the right way is use this valid_paths as keys to search in s3, and find all contents there for these keys as the valid entries to generate index file

Reply:
Yes I understand, but this one is only for local generation, and this is subject to change

Reply:
Patch sent, now it handles maven upload and overlap correctly